FirebaseCore/Sources/FIROptions.m

```diff
@@ -173,6 +173,12 @@ - (id)copyWithZone:(NSZone *)zone {
 
 #pragma mark - Public instance methods
 
+- (instancetype)init {
+  // Unavailable.
+  [self doesNotRecognizeSelector:_cmd];
```

Comment:
I wonder if we really need this? The initializer marked as unavailable should lead to a compiler error, so I guess it should be enough. Or there is a specific case we would like to cover?

Reply:
Ah, forgot to link! I was watching the "Refine Objective-C frameworks for Swift" WWDC session and that was the suggestion: https://developer.apple.com/videos/play/wwdc2020/10680/?time=1159

They suggested this to make sure there's no accidental calls there. It also surfaced the [`copyWithZone:` implementation](https://github.com/firebase/firebase-ios-sdk/pull/6633/files#diff-fb3d49a63a8c4afacada87c02bc8de7bR163) calling the `init` function when it should have likely called one of the designated initializers.